tests/data/instances/mc_tuples_with_background_instance_test.py

```diff
@@ -0,0 +1,284 @@
+                       'num_options': num_options}
+        self.indexed_instance.pad(max_lengths)
+        for answer_option_tuples in self.indexed_instance.answers_indexed:
+            for ans_tuple in answer_option_tuples:
+                assert len(ans_tuple) == num_slots
+
+        for background_tuple in self.indexed_instance.background_indexed:
+            assert len(background_tuple) == num_slots
+
+    def test_pad_add_words_in_slots(self):
+        num_question_tuples = 4
+        num_background_tuples = 4
+        num_slots = 4
+        question_slot_length = 30
+        background_slot_length = 40
+        num_options = 3
+        max_lengths = {'num_question_tuples': num_question_tuples,
+                       'num_background_tuples': num_background_tuples,
+                       'num_slots': num_slots,
+                       'question_slot_length': question_slot_length,
+                       'background_slot_length': background_slot_length,
+                       'num_options': num_options}
+        self.indexed_instance.pad(max_lengths)
+        for answer_option_tuples in self.indexed_instance.answers_indexed:
+            for ans_tuple in answer_option_tuples:
+                for slot in ans_tuple:
+                    assert len(slot) == question_slot_length
+
+        for background_tuple in self.indexed_instance.background_indexed:
+            for slot in background_tuple:
+                assert len(slot) == background_slot_length
+
+
+    def test_pad_remove_words_in_slots(self):
+        question_slot_length = 1
+        background_slot_length = 1
+        max_lengths = {'num_question_tuples': 4,
+                       'num_background_tuples': 4,
+                       'num_slots': 4,
+                       'question_slot_length': question_slot_length,
+                       'background_slot_length': background_slot_length,
+                       'num_options': 3}
+        self.indexed_instance.pad(max_lengths)
+        for answer_option_tuples in self.indexed_instance.answers_indexed:
+            for ans_tuple in answer_option_tuples:
+                for slot in ans_tuple:
+                    assert len(slot) == question_slot_length
+
+        for background_tuple in self.indexed_instance.background_indexed:
+            for slot in background_tuple:
+                assert len(slot) == background_slot_length
+
+    def test_as_training_data_produces_correct_numpy_arrays(self):
+        max_lengths = {'num_question_tuples': 2,
+                       'num_background_tuples': 3,
+                       'num_slots': 2,
+                       'question_slot_length': 2,
+                       'background_slot_length': 3,
+                       'num_options': 3}
+        self.indexed_instance.pad(max_lengths)
+
+        inputs, label = self.indexed_instance.as_training_data()
+        assert np.all(label == np.asarray([0, 0, 1]))
+
+        desired_options = np.asarray([[[[1, 2], [0, 3]],
```

Comment:
um... yes... so, with that tokenizer, the innermost list would not be a List[Int] (i.e. list of word indices, it would be a List[List[Int]], where each elem of the outer list is a list of indices represetning [word_index, char0_index, ..., charN_index] right?